nixos/modules/services/x11/display-managers/default.nix

```diff
@@ -126,6 +126,14 @@ let
         (*) echo "$0: Desktop manager '$desktopManager' not found.";;
       esac
 
+      ${optionalString cfg.startDbusSession ''
```

Comment:
A possibility would be to add an option `updateDbusEnvironment` that defaults to `false`, but can be set by desktop managers and users of more unusual environments if they require it. KDE5 should probably leave it `false` since it has its own method. For the Enlightenment, Gnome and XFCE, activating it should be ok (I only tested Gnome yet); not sure about Kodi and KDE4.